src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.domain.Fruit;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Fruit.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<Fruit.Operation, OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler get(Fruit.Operation operation) {
+        return operationHandlerMap.get(operation);
```

Comment:
When retrieving an operation handler, it's important to handle the case when the operation is not supported. Consider throwing a custom exception with an informative message if the operation handler is not found in the map.